File: filesync/src/main/java/pl/softfly/flashcards/filesync/db/FileSyncDeckDatabase.java
package pl.softfly.flashcards.filesync.db;

import androidx.room.Database;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

import java.util.List;

import pl.softfly.flashcards.dao.deck.CardDao;
import pl.softfly.flashcards.dao.deck.DeckConfigAsyncDao;
import pl.softfly.flashcards.db.Converters;
import pl.softfly.flashcards.entity.deck.Card;
import pl.softfly.flashcards.entity.deck.CardFts;
import pl.softfly.flashcards.entity.deck.CardLearningProgress;
import pl.softfly.flashcards.entity.deck.DeckConfig;
import pl.softfly.flashcards.entity.filesync.CardEdge;
import pl.softfly.flashcards.entity.filesync.CardImported;
import pl.softfly.flashcards.entity.filesync.CardImportedRemoved;
import pl.softfly.flashcards.entity.filesync.FileSynced;
import pl.softfly.flashcards.filesync.dao.FileSyncedDao;
import pl.softfly.flashcards.view.filesync.CountToGraphOnlyNewCards;
import pl.softfly.flashcards.view.filesync.GraphEdge;
import pl.softfly.flashcards.view.filesync.GraphEdgeOnlyNewCards;

/**
 * @author Grzegorz Ziemski
 */
@Database(
        entities = {
                Card.class,
                CardFts.class,
                CardLearningProgress.class,
                DeckConfig.class,
                CardImported.class,
                CardEdge.class,
                FileSynced.class,
                CardImportedRemoved.class
        },
        views = {
                GraphEdge.class,
                GraphEdgeOnlyNewCards.class,
                CountToGraphOnlyNewCards.class
        },
        exportSchema = false,
        version = 1
)
@TypeConverters({Converters.class})
public abstract class FileSyncDeckDatabase extends RoomDatabase {

    public abstract CardDao cardDao();

    public abstract FileSyncedDao fileSyncedDao();

    public abstract DeckConfigAsyncDao deckConfigAsyncDao();
}

File: filesync/src/main/java/pl/gocards/filesync/db/FileSyncDeckDatabase.kt
package pl.gocards.filesync.db

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import pl.gocards.db.Converters
import pl.gocards.room.entity.deck.*
import pl.gocards.room.entity.filesync.*
import pl.gocards.filesync.dao.CardDao
import pl.gocards.filesync.dao.FileSyncedDao

/**
 * @author Grzegorz Ziemski
 */
@Database(
    entities = [
        Card::class,
        CardFts::class,
        CardLearningHistory::class,
        CardLearningProgress::class,
        CardConfig::class,
        DeckConfig::class,
        CardImported::class,
        CardEdge::class,
        FileSynced::class,
        FileSyncedBackup::class,
        CardImportedRemoved::class
    ],
    views = [
        GraphEdge::class,
        GraphEdgeOnlyNewCards::class,
        CountToGraphOnlyNewCards::class
    ],
    exportSchema = false,
    version = 1
)
@TypeConverters(Converters::class)
abstract class FileSyncDeckDatabase : RoomDatabase() {
    abstract fun cardDao(): CardDao
    abstract fun fileSyncedDao(): FileSyncedDao
}